ontopy/ontology.py

```diff
@@ -353,23 +362,34 @@ def get_by_label(
 
         if colon_in_label is None:
             colon_in_label = self._colon_in_label
-        if colon_in_label:
-            if prefix:
-                raise ValueError(
-                    "`prefix` cannot be combined with `colon_in_label`"
+        if colon_in_label and prefix:
+            raise ValueError(
+                "`prefix` cannot be combined with `colon_in_label`"
+            )
+
+        splitlabel = label.split(":", 1)
+        if len(splitlabel) == 2 and not splitlabel[1].startswith("//"):
+            label = splitlabel[1]
+            if namespace and namespace != splitlabel[0]:
+                warnings.warn(
+                    f"Namespace given both as argument ({namespace}) "
+                    f"and in label as ({splitlabel[0]}). "
+                    "Namespace given in argument is ignored. "
                 )
-        else:
-            splitlabel = label.split(":", 1)
-            if len(splitlabel) == 2 and not splitlabel[1].startswith("//"):
-                label = splitlabel[1]
-                if prefix and prefix != splitlabel[0]:
-                    warnings.warn(
-                        f"Prefix given both as argument ({prefix}) "
-                        f"and in label ({splitlabel[0]}). "
-                        "Prefix given in argument takes precedence. "
-                    )
-                if not prefix:
-                    prefix = splitlabel[0]
+            namespace = splitlabel[0]
+
+        if namespace:
+            entityset = self.get_by_label_all(
```

Comment:
Do you really want to call `get_by_label_all()` from within `get_by_label()`?
`get_by_label_all()` does wildcard matching. Do we really want that in `get_by_label()`?
Wouldn't it make more sense that `get_by_label()` does fast exact matching?